src/lib/InputMapper/SearchQueryMapper.php

```diff
@@ -129,12 +129,8 @@ static function ($sortClauseClass) {
      *
      * @return \Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\DateMetadata[]
      */
-    private function mapDateMetadata(array $queryArg = [], $dateMetadata)
+    private function mapDateMetadata(array $queryArg, $dateMetadata)
     {
-        if (!isset($queryArg[$dateMetadata]) || !is_array($queryArg[$dateMetadata])) {
-            return [];
-        }
```

Comment:
Are you sure this is supposed to be removed? This does not seem right.

Reply:
I can't say what will break exactly if we do this, but it indeed doesn't seem right.

If it's about the default value for a non-last argument, it sounds safer to make `$dateMetadata` optional as well. What do you think ?

Reply:
It's not about the first argument presence, but it's structure. `$queryArg['Modified']` and `$queryArg['Created']` (because this private method is called with this configuration only) may or may not be present, as the `$queryArg` parameter is actually coming from outside of this class.

Furthermore, this method argument list can be simplified, because what actually happens here is that the first argument is the array, while the second is the key from that array that data is read from. So we could just reduce it to that array part instead.

Reply:
Indeed, let's keep this check as it was before. Given, we aim to get rid of the deprecation warning only, I think making the first argument mandatory will do. AFAICS this method is called only twice, in both cases the first argument is present.